File: packages/dart/lib/src/territorial/generate_territorial_summary.dart
import 'intersection.dart';

const List<String> _order = ['ALTO', 'MEDIO', 'BAJO'];

String generateTerritorialSummary(List<Intersection> intersections) {
  if (intersections.isEmpty) {
    return 'El territorio no presenta superposiciones con las capas evaluadas.';
  }

  String dominant = 'BAJO';
  for (final level in _order) {
    if (intersections.any((i) => i.level == level)) {
      dominant = level;
      break;
    }
  }

  final entities = intersections.map((i) => i.entity).join(', ');
  final n = intersections.length;

  return 'El territorio presenta superposicion de nivel $dominant con $n '
      'entidad(es): $entities.';
}

File: packages/rust/src/territorial/generate_territorial_summary.rs
use super::types::Intersection;

pub fn generate_territorial_summary(intersections: &[Intersection]) -> String {
    if intersections.is_empty() {
        return "El territorio no presenta superposiciones con las capas evaluadas.".to_string();
    }

    let dominant = if intersections.iter().any(|i| i.level == "ALTO") {
        "ALTO"
    } else if intersections.iter().any(|i| i.level == "MEDIO") {
        "MEDIO"
    } else {
        "BAJO"
    };

    let entities: Vec<&str> = intersections.iter().map(|i| i.entity.as_str()).collect();

    format!(
        "El territorio presenta superposicion de nivel {} con {} entidad(es): {}.",
        dominant,
        intersections.len(),
        entities.join(", ")
    )
}
